### PR TITLE
Enable ESC and clicking background to close full screen dialog

### LINKED ISSUE #7904: Admin Page: no option to dismiss Warm Welcome
Warm Welcome was added in #7837.

Unfortunately the screen cannot be dismissed. These are the only actions one can take.

screenshot 2017-09-29 at 23 20 12

Could we add an option to dismiss?

### PR DESCRIPTION
**Pull request by gravityrail**
Fixes #7904

#### Changes proposed in this Pull Request:

* Clicking the background or pressing the ESC key should close full screen dialogs
* Move close button into the dialog body.

close-button-moved

#### Testing instructions:

* Trigger a full-screen dialog. This code snippet in a micro-plugin will do the trick for forcing an upgrade prompt on any of the Jetpack dash/settings pages:

```php
add_action( 'plugins_loaded', function() {
    Jetpack::state( 'message', 'modules_activated' );
});
```

* Click the background, the dialog should be closed
* Click the foreground (main part of dialog or picture above it) - the dialog should NOT be closed
* Re-open the dialog (e.g. by re-loading). Hit the ESC key. The dialog should be closed.

This also moves the close button into the top-right corner of the dialog, and includes an updated gridicons file which includes the circular close icon. Other uses of gridicons should be re-tested to ensure I didn't break anything.

#### Proposed changelog entry for your changes:

Better usability for full-screen dialogs.

### CODE REVIEW

File: _inc/client/components/jetpack-dialogue/index.jsx [hunk 1]
```diff
@@ -17,13 +17,31 @@ import onKeyDownCallback from 'utils/onkeydown-callback';
 import { imagePath } from 'constants/urls';
 
 class JetpackDialogue extends Component {
+	clickBackground( e ) {
+		this.props.dismiss( e );
+	}
+
+	// capture the ESC key globally
+	componentDidMount(){
+		document.addEventListener('keydown', this.props.dismiss, false);
```
Comment (dereksmart) [inline _inc/client/components/jetpack-dialogue/index.jsx]: I think these new rules should respect `this.props.showDismiss` -- there may be instances we don't want to allow dismissal without action.

Comment (gravityrail) [reply _inc/client/components/jetpack-dialogue/index.jsx]: Good idea - fixed

Comment (dereksmart) [reply _inc/client/components/jetpack-dialogue/index.jsx]: `git push`? While you're in there can you apply code standard spacing?

**Comment by dereksmart** [comment]
`esc` is throwing a console error with the recent changes

**Comment by dereksmart** [comment]
Is the update to the gridicon library necessary for this change?  I think it will require some more testing.  These changes now are modifying existing styles, example:

screen shot 2018-01-31 at 5 10 36 pm

**Comment by dereksmart** [comment]
`yarn lint` is throwing some a11y errors causing the tests to fail 

screen shot 2018-01-31 at 5 14 12 pm

**Comment by joanrho** [comment]
@gravityrail can you replace the "cross-circle" Gridicon with either the "cross" or "cross-small" instead?

**Comment by gravityrail** [comment]
@joanrho - icon updated, and screenshot in description

@dereksmart - feedback addressed, going to dismiss your review so you can take another look :)

File: _inc/client/components/jetpack-dialogue/index.jsx (hunk 1 above)
Comment (gravityrail) [reply _inc/client/components/jetpack-dialogue/index.jsx]: Fixed

**Comment by joanrho** [comment]
Design review approved. Thanks for the edits, Dan!

**Comment by zinigor** [comment]
Ported to `branch-5.8` in 36b4d680a.

**Pull request merged**